### PR TITLE
Adds handling for empy task and course item  page

### LINKED ISSUE #766: Do not let the item page empty
## Overview
For instance on https://dev.algorea.org/en/#/activities/by-id/2268458803184278714;path=;attempId=0/details (with the usual user), we get:
Screenshot 2021-10-15 at 16 35 37

Which is not very user friendly. :-)

## Acceptance criteria

For task/courses:

- [x] if there is no `url` (empty, null, undefined): 
   - [x] Display the error "This activity has not been correctly configured."
   - [x] if the current user can edit the item, add "You need to set a url in editing mode."

For any item type: (excluding the case above)

If no "current result": 
- [x] and `requires_explicit_entry=1` display the error:  "This activity requires explicit entry (not implemented)." 
- [x] and `requires_explicit_entry=0` display the error:  "Starting the activity ... (if you are stuck on this message, please retry and contact us if the problem persists)" 

Provide scenarios (link to items) so that the reviewer can check each case easily. (probably the last case cannot be reproduced)

### PR DESCRIPTION
**Pull request by Iloveall**
## Description

Fixes #766

## Notes (out of scope, known isues, hints for reviewing code, ...)  (optional)

...

## Test cases

- [x] Case 1:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/do-not-let-item-page-empty/en/#/activities/by-id/2268458803184278714;path=;attempId=0/details)
  4. Then I see "This activity has not been correctly configured" and "You need to set a url in editing mode." messages
  

- [x] Case 2:
  1. Given am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/do-not-let-item-page-empty/en/#/activities/by-id/4161175831988689371;path=1352246428241737349,561969155071897034;parentAttempId=0/details)
  4. Then I see Chapter with "This activity requires explicit entry (not implemented)." message

- [x] Case 3:
  1. Given am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/do-not-let-item-page-empty/en/#/activities/by-id/7484937021725214233;path=1352246428241737349,561969155071897034;parentAttempId=0/details)
  4. Then I see Task with "This activity requires explicit entry (not implemented)." message
  
- [x] Case 4:
  1. Given am the user demouser
  2. When I go to [this page](https://dev.algorea.org/branch/feature/do-not-let-item-page-empty/en/#/activities/by-id/1501663083087440078;path=4702,1625159049301502151;parentAttempId=0/details)
  4. Then I see Task with "This activity has not been correctly configured." message

### CODE REVIEW

File: src/app/modules/item/pages/item-content/item-content.component.html [hunk 1]
```diff
@@ -1,23 +1,36 @@
 <ng-container *ngIf="itemData">
```
[the remark this answers is not in the document]
Comment (Iloveall) [reply src/app/modules/item/pages/item-content/item-content.component.html:1]: Yes, we have agreement with @smadbe to improve it in next task.

**Review by ThomasJuster — COMMENTED** [review]
Good to go for me :heavy_check_mark: 
@smadbe since it's your issue … your turn :grin: 

I could also test the use case `display "Starting the activity ..."` when visiting [this chapter](https://dev.algorea.org/branch/feature/do-not-let-item-page-empty/en/#/activities/by-id/1625159049301502151;path=4702;attempId=0/details) as usual user by throttling network to bad connection like 3G
Since the result is built up -on item in datasource, this message is displayed while the item is fetched **and** the results are being fetched. As a user I would have expected a loader, but when reading the issue I understand it as the expected behavior, is it ?

**Review by smadbe — CHANGES_REQUESTED** [review]
* Could you merge into this branch and fix the conflict first?  As your code is pretty old, item-content has changed in between.

* This course with explicit entry should behave like others: https://dev.algorea.org/branch/feature/do-not-let-item-page-empty/en/#/activities/by-id/596025909122824879;path=1352246428241737349,561969155071897034;attempId=0/details

* Could you provide a test case of a task without url but on which the user has no editing permissions?

* as suggested by Thomas, please add a loader after the "Starting the activity ... " message.

**Review by ThomasJuster — COMMENTED** [review]
I retested and checked the loader, works fine :+1: 
I'm leaving the rest to Damien

**Review by smadbe — APPROVED** [review]

**Pull request merged**